### PR TITLE
D store dev POST method functionality

### PR DESCRIPTION
**Pull request by jpaints3**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/ContactMeFormSubmissionServlet.java [hunk 1]
```diff
@@ -1,16 +1,44 @@
...
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
+      // get form information
+      String name = Jsoup.clean(request.getParameter("name"), Whitelist.none());
+      String email = Jsoup.clean(request.getParameter("email"), Whitelist.none());
+      String reason = Jsoup.clean(request.getParameter("reason"), Whitelist.none());
```
Comment (josueetcom) [inline portfolio/src/main/java/com/google/sps/servlets/ContactMeFormSubmissionServlet.java:24]: Optional: In newer versions of `jsoup`, [`Whitelist`](https://jsoup.org/apidocs/org/jsoup/safety/Whitelist.html) is deprecated in favor of `Safelist`. Feel free to use `Safelist` if you prefer (though I think it requires upgrading the version of jSoup in your pom.xml to at least 1.15.1 so I'm not too sure it works).

File: portfolio/src/main/java/com/google/sps/servlets/ContactMeFormSubmissionServlet.java [hunk 2]
```diff
@@ -1,16 +1,44 @@
...
+
+      // add kind submissions to keyfactory
+      keyFactory.setKind("submissions");
+     
+      //take the submission elements and create a new entity object for it 
+      FullEntity submissionEntity =
+        Entity.newBuilder(keyFactory.newKey())
+            .set("e-name", name)
+            .set("e-email", email)
+            .set("e-reason", reason)
```
Comment (josueetcom) [inline portfolio/src/main/java/com/google/sps/servlets/ContactMeFormSubmissionServlet.java:38]: Interesting `e-` prefixes 🤔

File: portfolio/src/main/webapp/submit.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+//fetch function for submission servlet
+async function submissionFetch() {
+  const completionMessage = await fetch("/submitform");
+
+  document.getElementById("postComplete").innerText = completionMessage;
```
Comment (josueetcom) [inline portfolio/src/main/webapp/submit.js:5]: "Use single quotes"

https://google.github.io/styleguide/jsguide.html#features-strings-use-single-quotes

File: portfolio/src/main/webapp/submit.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+//fetch function for submission servlet
```
Comment (josueetcom) [inline portfolio/src/main/webapp/submit.js:1]: Need a space after `//`

https://google.github.io/styleguide/jsguide.html#formatting-horizontal-whitespace

File: portfolio/src/main/webapp/submit.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+async function submissionFetch() {
+  const completionMessage = await fetch("/submitform");
```
Comment (josueetcom) [inline portfolio/src/main/webapp/submit.js:3]: I believe `fetch()` by default does a `GET` request. It looks like you don't have a `doGet` to handle it. Also, should this be `/submitForm`?

File: portfolio/src/main/java/com/google/sps/servlets/ContactMeFormSubmissionServlet.java (hunk 2 above)
Comment (jpaints3) [reply portfolio/src/main/java/com/google/sps/servlets/ContactMeFormSubmissionServlet.java:38]: yea, I wasn't sure if I should make it known those are the entity categories 🤔

**Pull request merged**